src/System.Private.CoreLib/src/System/Runtime/Loader/ComponentDependencyResolver.cs

```diff
@@ -0,0 +1,282 @@
+            {
+                returnCode = corehost_resolve_component_dependencies(
+                    componentAssemblyPath,
+                    (assembly_paths, native_search_paths, resource_search_paths) =>
+                    {
+                        assemblyPathsList = assembly_paths;
+                        nativeSearchPathsList = native_search_paths;
+                        resourceSearchPathsList = resource_search_paths;
+                    });
+            }
+            catch (EntryPointNotFoundException entryPointNotFoundException)
+            {
+                throw new InvalidOperationException(SR.ComponentDependencyResolver_FailedToLoadHostpolicy, entryPointNotFoundException);
+            }
+            catch (DllNotFoundException dllNotFoundException)
+            {
+                throw new InvalidOperationException(SR.ComponentDependencyResolver_FailedToLoadHostpolicy, dllNotFoundException);
+            }
+
+            if (returnCode != 0)
+            {
+                // Something went wrong - report a failure
+                // TODO - once we have the ability to capture detailed error messages from the hostpolicy
+                // use it here to make the exception more actionable.
+                throw new InvalidOperationException(SR.Format(SR.ComponentDependencyResolver_FailedToResolveDependencies, componentAssemblyPath, returnCode));
+            }
+
+            string[] assemblyPaths = SplitPathsList(assemblyPathsList);
+
+            // Assembly simple names are case insensitive per the runtime behavior
+            // (see SimpleNameToFileNameMapTraits for the TPA lookup hash).
+            _assemblyPaths = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (string assemblyPath in assemblyPaths)
+            {
+                _assemblyPaths.Add(Path.GetFileNameWithoutExtension(assemblyPath), assemblyPath);
+            }
+
+            _nativeSearchPaths = SplitPathsList(nativeSearchPathsList);
+            _resourceSearchPaths = SplitPathsList(resourceSearchPathsList);
+
+            _componentDirectorySearchPaths = new string[1] { Path.GetDirectoryName(componentAssemblyPath) };
+        }
+
+        public string ResolveAssemblyPath(AssemblyName assemblyName)
+        {
+            // Determine if the assembly name is for a satellite assembly or not
+            // This is the same logic as in AssemblyBinder::BindByTpaList in CoreCLR
+            // - If the culture name is non-empty and it's not 'neutral' 
+            // - The culture name is the value of the AssemblyName.Culture.Name 
+            //     (CoreCLR gets this and stores it as the culture name in the internal assembly name)
+            //     AssemblyName.CultureName is just a shortcut to AssemblyName.Culture.Name.
+            if (!string.IsNullOrEmpty(assemblyName.CultureName) && 
+                !string.Equals(assemblyName.CultureName, "neutral", StringComparison.OrdinalIgnoreCase))
```

Comment:
I haven't seen the CultureName == "neutral" when looking at resolving event.  Maybe there are code paths I haven't looked at.  I wonder if this is dead native code.